File: ruby/easy/reverse_words_in_a_string_iii.rb
# @param {String} s
# @return {String}
def reverse_words(s)
    res = word = ""
    s.each_char.with_index do |c, i|
        if(c == ' ')
            res += reverse(word)
            res += ' ' if i > 0 && i < s.length-1
            word = ""
        else
            word += c
        end
    end
    return res + reverse(word)
end

def reverse(word)
    reverse = ""
    j = word.length-1
    while(j >= 0)
        reverse += word[j]
        j -= 1
    end
    return reverse
end

File: javascript/easy/reverseWordsInAStringIII.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    let res = "", word = "";
    for(let i = 0; i < s.length; i++) {
        if(s[i] === ' ') {
            res += reverseWord(word) + ((i > 0 && i < s.length-1) ? ' ' : '');
            word = "";
        } else {
            word += s[i];
        }
    }
    return res + reverseWord(word);
};

var reverseWord = function(word) {
    let res = "";
    for(let i = 0; i < word.length; i++) {
        res = word[i] + res;
    }
    return res;
};
